### ISSUE TITLE
[2020-06-29][박진영] Client 개발 태스크 분배 및 일정논의 + Bubble UI 관련 컴포넌트 제작중

### ISSUE THREAD

**Issue by ed-jinyoung-park** [boogle-lab/ssho-daily-scrum-board#2]
### 개발 태스크 분배 및 일정 논의
(링크 : https://www.notion.so/0629-2b8d135962fd4576a382f67b257d57a5)

**주요 논의 사항**
- 화면 단위로 역할을 나누어 각자가 맡은 화면을 완성하는 방식으로 합의.
- 매주 Android, ios 디바이스로 테스팅
- scrum board에 그날의 한 일을 공유하되, 프로젝트의 readme에 설치환경, 써드파티 라이브러리 등을 명세.

**1-4주차 계획**
- 유진님 - 캐로우젤을 포함한 스와이프 뷰 구현
- 진영 - 버블 뷰 구현

### Bubble UI 컴포넌트 제작을 위한 레퍼런스 공유
(레퍼런스 : https://github.com/jesster2k10/react-native-bubble-select)
![image](https://user-images.githubusercontent.com/26531678/86013322-fd22e680-ba59-11ea-9c3c-3cc7f57709cf.png)

- 라이브러리를 import하는 방식 대신 커스터마이징을 쉽게 하기 위해 소스코드를 참조하여 직접 컴포넌트 제작 예정.
- BubbleSelectView 안에 Bubble Component가 여러개 있는 식으로 구성되어 있음.

### 프로젝트 시작 시 react native init vs expo init 관련
react native로 프로젝트 시작시 위의 두 가지 방법이 있습니다.

react native init은 초기 개발 환경 세팅 및 관리 운영이 어려우나 필요한 모듈을 만들어 도입할 수 있다는 장점이 있습니다.

반면 expo init은 초기 개발 환경 세팅이 쉬우나, 빌드 파일 크기가 매우 크며 모듈 제작이 불가능하여 규모가 크고 기능이 많은 앱을 개발하는 것은 어려움이 있습니다.

현재 프로토타입의 경우 모듈을 만들거나 환경 세팅을 고칠 필요가 없으므로 합의한 바와 같이 **expo를 통해** 개발하는 것이 적절할 것으로 생각하며, 향후 프로덕트 개발 시에는 react native init 도입을 검토해보는 것이 필요하다고 생각합니다.

**Comment by ywoo-park** [comment]
@Zinyon 하반기에 제품 레벨의 개발을 할 때는 react-native-cli 기반으로 프로젝트 구성 및 앱 빌드를 진행하는 것이 좋아보입니다!

expo 기반의 프로젝트 구성이 

1. 빌드 파일이 크다
2. native 코드(java, kotlin, swift, objective-c 등을 작성 할 수 없다
3. 앱 빌드가 expo가 관리하는 빌드 작업 큐 내에서 진행되어, 배포에 걸리는 시간이 길다

정도의 단점이 있는것 같습니다!

이번주에 스와이프 / 버블 custom 구현과 관련해서 알아보실 때 native로 작성해야 하는 부분이 있을지 고려해보시면 좋을 것 같습니다!

**Comment by ed-jinyoung-park** [comment]
네 안그래도 버블 관련해서 찾은 라이브러리를 직접 구현하려면 native로 작성해야 하는 부분이 필요해서 1)그냥 import해서 구현할지 2) react native만으로 한번 만들어볼지 고민중입니다!
